weldx/geometry.py

```diff
@@ -698,98 +810,32 @@ def apply_transformation(self, matrix: np.ndarray):
-
-    @UREG.wraps(_DEFAULT_LEN_UNIT, (None, _DEFAULT_LEN_UNIT), strict=True)
-    def rasterize(self, raster_width: pint.Quantity) -> pint.Quantity:
-        """Create an array of points that describe the segments contour.
-
-        The effective raster width may vary from the specified one,
-        since the algorithm enforces constant distances between two
-        raster points.
-
-        Parameters
-        ----------
-        raster_width :
-            The desired distance between two raster points
-
-        Returns
-        -------
-        pint.Quantity
-            Array of contour points
-
-        """
-        point_start = self.point_start.m
-        point_center = self.point_center.m
-        vec_center_start = point_start - point_center
-        if not raster_width > 0:
-            raise ValueError("'raster_width' must be > 0")
-        raster_width = np.clip(raster_width, None, self.arc_length.m)
-
-        num_raster_segments = int(np.round(self._arc_length / raster_width))
-        delta_angle = self._arc_angle / num_raster_segments
-
-        max_angle = self._sign_arc_winding * (self._arc_angle + 0.5 * delta_angle)
-        angles = np.arange(0, max_angle, self._sign_arc_winding * delta_angle)
+        self._points = np.matmul(np.array(matrix), self._points)
+        self._sign_winding *= tf.reflection_sign(np.array(matrix))
 
-        rotation_matrices = tf.WXRotation.from_euler("z", angles).as_matrix()[
-            :, 0:2, 0:2
-        ]
-
-        data = np.matmul(rotation_matrices, vec_center_start) + point_center
-
-        return data.transpose()
-
-    def transform(self, matrix) -> ArcSegment:
-        """Get a transformed copy of the segment.
-
-        Parameters
-        ----------
-        matrix :
-            Transformation matrix
-
-        Returns
-        -------
-        ArcSegment
-            Transformed copy
+        self._series = self._update_internals_and_get_series()
+        self._update_internals()
+        self._check_valid()
 
-        """
-        new_segment = copy.deepcopy(self)
-        new_segment.apply_transformation(matrix)
-        return new_segment
+        return self
 
-    @UREG.check(None, "[length]")
-    def translate(self, vector) -> ArcSegment:
-        """Get a translated copy of the segment.
+    @UREG.check(None, _DEFAULT_LEN_UNIT)
+    def apply_translation(self, vector):
+        """Apply a translation in place.
 
         Parameters
         ----------
-        vector :
-            Translation vector
+        vector:
+            The translation vector.
 
         Returns
         -------
-        ArcSegment
-            Transformed copy
+        DynamicShapeSegment:
+            A self-reference to the modified segment
 
         """
-        new_segment = copy.deepcopy(self)
-        new_segment.apply_translation(vector)
-        return new_segment
+        self._points = (self._points.transpose() + vector.m).transpose()
```

Comment:
Is `self._points` not a quantity? I'm a little confused by the `.magnitude` here

Reply:
The internal format is without a unit. The unit is removed by `ureg.wraps`. I'll check this line again.

Reply:
I think `@UREG.check` doesn't remove units like `@UREG.wraps` would

Reply:
No it doesn't, but the `__init__` currently uses `wraps`. I can check if changing that to `check` is an easy fix.

Reply:
I now changed it and it seems to work. Now some internals are quantities and the usage of the magnitude could be further reduced